components/director/internal/open_resource_discovery/validation.go

```diff
@@ -604,6 +626,42 @@ func validateEventInputWithSuppressedErrors(event *model.EventDefinitionInput, e
 		})))
 }
 
+func validateCapabilityInput(capability *model.CapabilityInput) error {
+	return validation.ValidateStruct(capability,
+		validation.Field(&capability.OrdPackageID, validation.Required, validation.Match(regexp.MustCompile(PackageOrdIDRegex))),
+		validation.Field(&capability.Name, validation.Required),
+		validation.Field(&capability.Description, validation.Required, validation.Length(MinDescriptionLength, MaxDescriptionLength)),
+		validation.Field(&capability.OrdID, validation.Required, validation.Match(regexp.MustCompile(CapabilityOrdIDRegex))),
+		validation.Field(&capability.Type, validation.Required, validation.In(CapabilityTypeCustom, CapabilityTypeMDICapabilityDefinitionV1), validation.When(capability.CustomType != nil, validation.In(CapabilityTypeCustom))),
+		validation.Field(&capability.CustomType, validation.When(capability.Type != CapabilityTypeCustom, validation.Empty), validation.Match(regexp.MustCompile(CapabilityCustomTypeRegex))),
+		validation.Field(&capability.LocalTenantID, validation.NilOrNotEmpty, validation.Length(MinLocalTenantIDLength, MaxLocalTenantIDLength)),
+		validation.Field(&capability.ShortDescription, shortDescriptionRules...),
+		validation.Field(&capability.Tags, validation.By(func(value interface{}) error {
+			return validateJSONArrayOfStringsMatchPattern(value, regexp.MustCompile(StringArrayElementRegex))
+		})),
+		validation.Field(&capability.Links, validation.By(validateORDLinks)),
+		validation.Field(&capability.ReleaseStatus, validation.Required, validation.In(ReleaseStatusBeta, ReleaseStatusActive, ReleaseStatusDeprecated)),
+		validation.Field(&capability.Labels, validation.By(validateORDLabels)),
+		validation.Field(&capability.Visibility, validation.Required, validation.In(CapabilityVisibilityPublic, CapabilityVisibilityInternal, CapabilityVisibilityPrivate)),
+		validation.Field(&capability.CapabilityDefinitions, validation.By(func(value interface{}) error {
```

Comment:
I think this is what we need - `In case that this property is available and the value of it hasn't changed since the last crawling, the resource definitions do not need to be fetched and updated.`

The requirement which i mentioned firstly does not concern us!

Reply:
Adjusted `if` statement to use `lastUpdate` in resyncAPI, resyncEvent and resyncCapability to determine if specs should be fetched.